homeassistant/components/modem_callerid/sensor.py

```diff
@@ -34,7 +35,20 @@
 )
 
 
-async def async_setup_entry(hass, entry, async_add_entities):
+async def async_setup_platform(
+    hass: HomeAssistant, config: ConfigEntry, async_add_entities, discovery_info=None
```

Comment:
If we type anything we should type the whole function:

`async_add_entities: AddEntitiesCallback`

`from homeassistant.helpers.entity_platform import AddEntitiesCallback`

`discovery_info: DiscoveryInfoType | None = None`

`from homeassistant.helpers.typing import DiscoveryInfoType`

Also add a return type